src/modm/platform/spi/stm32/spi_master_dma_impl.hpp.in

```diff
@@ -105,6 +105,72 @@ modm::platform::SpiMaster{{ id }}_Dma<DmaChannelRx, DmaChannelTx>::transfer(cons
 		// we will only visit this state once
 		state |= Bit1;
 		dmaError = false;
+		
+		Dma::TxChannel::template configureDataSize<T>();
```

Comment:
But i've kept the templatized `transfer(...)` / plain overloads for ATmega.

because:
1. It feels right to control types using template arguments. Always. (+++points for generics)
2. On calling transfer(...), [T should be deduced](https://en.cppreference.com/w/cpp/language/template_argument_deduction). Some drivers need an explicit typecast for there data arg now.
3. The DMA-channel configuration still needs to know about T, so i can **configure the DataSize**. It's conciderable to add a ConfigurationHandler support to the DMA APIs as well.

Reply:
> The DMA-channel configuration still needs to know about T, so that the DataSize can be configured 🙄.

This should be set in `Spi::setDataWidth`. There are only two types `uint8_t` and `uint16_t`.